### PR TITLE
Ability to delete multiple jobs by passing multiple ids

### PR DESCRIPTION
**Pull request by AleksandrMikhin**
This PR brings an ability to delete jobs by passing multi IDs.

Short description:
- implemented an ability to delite jobs by passing multi IDs
- changed the `DeletedJobInfo` class and the job deletion response
- fixed obtaining an owner of a job when it's deleted [SGE]

additionally:
- made tests refactoring of class `SgeJobProviderTest` class
- improved validation of ParallelEnvOptions passed to job submitting requests [SGE]
- added gradle task to run single tests from IDE without filtration

### CODE REVIEW

File: src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java [hunk 1]
```diff
@@ -161,19 +159,48 @@ public CommandType getProviderType() {
...
-        return buildNewJob(getResultOfExecutedCommand(simpleCmdExecutor, makeQsubCommand(options)));
+        return buildNewJob(toSubmitNewJob(options));
+    }
+
+    private long toSubmitNewJob(final JobOptions options) {
+        final CommandResult result = simpleCmdExecutor.execute(makeQsubCommand(options));
+        if (result.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(result, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        final Matcher matcher = SUBMITTED_JOB_ID_PATTERN.matcher(result.getStdOut().get(0));
```
Comment (SilinPavel) [inline src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java]: possible NPE here, when `get(0)` is called

File: src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java [hunk 2]
```diff
@@ -236,85 +255,50 @@ private String[] makeQdelCommand(final DeleteJobFilter filter) {
      * @param deleteJobFilter User-defined conditions.
      */
     private void validateDeleteRequest(final DeleteJobFilter deleteJobFilter) {
-        if (!StringUtils.hasText(deleteJobFilter.getUser()) && deleteJobFilter.getId() == null) {
+        if (!StringUtils.hasText(deleteJobFilter.getUser()) && CollectionUtils.isEmpty(deleteJobFilter.getIds())) {
```
Comment (SilinPavel) [inline src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java]: how system will behave if we pass `owner` as well as `ids`, but ids weren't launched from `owner`? I believe it is better to check it here to, and do not allow run request if both things are specified

Comment (SilinPavel) [reply src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java]: I also believe that such check could be moved to service layer to reuse this code

**Review by SilinPavel — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java (hunk 1 above)
Comment (AleksandrMikhin) [reply src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java]: fixed the if-condition

File: src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java (hunk 2 above)
Comment (AleksandrMikhin) [reply src/main/java/com/epam/grid/engine/provider/job/sge/SgeJobProvider.java]: done

File: src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java [hunk 3]
```diff
@@ -80,26 +81,30 @@ public Listing<Job> filter(final JobFilter jobFilter) {
...
+            if (CollectionUtils.isEmpty(deleteJobFilter.getIds())) {
                 throw new GridEngineException(HttpStatus.BAD_REQUEST, String.format("Incorrect filling in %s. "
-                                + "Either `id` or `user` should be specified for job removal!", deleteJobFilter));
-            }
-            if (deleteJobFilter.getId() <= 0) {
-                throw new GridEngineException(HttpStatus.BAD_REQUEST,
-                        String.format("Id specified in %s for job removal is invalid!", deleteJobFilter));
+                        + "Either at least one `id` or `user` must be specified to delete jobs!", deleteJobFilter));
             }
+            deleteJobFilter.getIds().stream()
```
Comment (ekazachkova) [inline src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java]: It looks this filter can be simplified: `.stream().filter(id -> id == null ||  id <= 0)`. Could you please check it?

File: src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java [hunk 4]
```diff
@@ -197,487 +152,237 @@ public class SgeJobProviderTest {
...
-        final List<String> hostList = Collections.singletonList(TWO_VALID_XML);
-        final JobFilter jobFilter = new JobFilter();
-        jobFilter.setNames(NAME);
-        final Job runningJob = runningJobTemplate();
-        final CommandResult commandResult = CommandResult.builder()
-                .stdOut(hostList)
-                .stdErr(EMPTY_LIST)
-                .build();
-
+    public void shouldTrowWhenCommandResultFiltrationWithErrorCode() {
```
Comment (ekazachkova) [inline src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java]: typo `Trow` -> `Throw`

File: src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java [hunk 5]
```diff
@@ -197,487 +152,237 @@ public class SgeJobProviderTest {
...
+    public void shouldThrowExceptionBecauseParallelExecOptionsAreUsed() {
+        final JobOptions jobOptions = JobOptions.builder()
+                .command(JOB_COMMAND)
+                .parallelExecutionOptions(new ParallelExecutionOptions())
                 .build();
+        Assertions.assertThrows(UnsupportedOperationException.class,
+                () -> sgeJobProvider.runJob(jobOptions, SOME_JOB_LOG_DIRECTORY_PATH));
     }
 
     private static Job runningJobTemplate() {
```
Comment (ekazachkova) [inline src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java]: Let's move this utility private methods to the end of the file

File: src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java (hunk 4 above)
Comment (AleksandrMikhin) [reply src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java]: done

File: src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java (hunk 3 above)
Comment (AleksandrMikhin) [reply src/main/java/com/epam/grid/engine/service/JobOperationProviderService.java]: done

File: src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java (hunk 5 above)
Comment (AleksandrMikhin) [reply src/test/java/com/epam/grid/engine/provider/job/sge/SgeJobProviderTest.java]: moved

**Pull request merged**